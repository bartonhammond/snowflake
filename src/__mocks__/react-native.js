/**
 * # __mockes__/react-native.js
 * 
 * This class stubs out the React-Native classes with React classes
 */
'use strict';
/**
 * ## Imports
 * 
 * ReactNative is actually React
 */ 
const React = require('react');
const ReactNative = React;

/**
 * ## These need additional mocking
 * 
 * ReactNative is actually React
 */ 
ReactNative.StyleSheet = {
    create: function create(styles) {
        return styles;
    }
};
class View extends React.Component {
    render() { return false; }
}
class PixelRatio extends React.Component {
    static get() { return 1; }
}
/**
 * ## Stubs
 * 
 * Simple replacements for testing
 */ 
ReactNative.View = View;
ReactNative.ScrollView = View;
ReactNative.Text = View;
ReactNative.TouchableOpacity = View;
ReactNative.TouchableHighlight = View;
ReactNative.TouchableWithoutFeedback = View;
ReactNative.ToolbarAndroid = View;
ReactNative.Image = View;
ReactNative.PixelRatio = PixelRatio;
ReactNative.NativeModules= {};

ReactNative.Platform = {};
module.exports = ReactNative;
