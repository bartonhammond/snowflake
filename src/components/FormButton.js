/**
 * # FormButton.js
 *
 * Display a button that responds to onPress and is colored appropriately
 */
'use strict'
/**
 * ## Imports
 *
 * React
 */
import React from 'react'
import
{
  StyleSheet,
  View
} from 'react-native'

/**
 * The platform neutral button
 */
const Button = require('apsl-react-native-button')

/**
 * ## Styles
 */
var styles = StyleSheet.create({
  signin: {
    marginLeft: 10,
    marginRight: 10
  },
  button: {
    backgroundColor: '#FF3366',
    borderColor: '#FF3366'
  }

})

var FormButton = React.createClass({
  /**
   * ### render
   *
   * Display the Button
   */
  render () {
    return (
      <View style={styles.signin}>
        <Button style={styles.button}
          textStyle={{fontSize: 18}}
          isDisabled={this.props.isDisabled}
          onPress={this.props.onPress} >
          {this.props.buttonText}
        </Button>
      </View>
    )
  }
})

module.exports = FormButton
