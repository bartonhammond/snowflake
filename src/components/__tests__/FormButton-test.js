/**
 * # FormButton-test.js
 *
 * This class tests that the form button displays correctly
 *
 * *Note:* if you want to understand the structures here, add a
 * ```console.log``` and then ```npm test```.
 */
'use strict'

/**
* ## Imports
 */

import 'react-native'
import React from 'react'

import FormButton from '../FormButton'

import ReactTestUtils from 'react-addons-test-utils'
const renderer = ReactTestUtils.createRenderer()

it('FormButton', () => {
  const props = {
    isDisabled: false,
    onPress: () => {},
    buttonText: 'TestString'
  }
  renderer.render(<FormButton {...props} />)
  const tree = renderer.getRenderOutput()
  expect(tree).toMatchSnapshot()
})
