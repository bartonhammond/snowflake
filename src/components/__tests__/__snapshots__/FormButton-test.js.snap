exports[`test FormButton 1`] = `
<View
  style={
    Object {
      "marginLeft": 10,
      "marginRight": 10,
    }
  }>
  <Button
    isDisabled={false}
    onPress={[Function]}
    style={
      Object {
        "backgroundColor": "#FF3366",
        "borderColor": "#FF3366",
      }
    }
    textStyle={
      Object {
        "fontSize": 18,
      }
    }>
    TestString
  </Button>
</View>
`;
